elks/arch/i86/drivers/char/console-direct.c

```diff
@@ -80,38 +93,38 @@ static void std_char(register Console *, int);
 
 static void SetDisplayPage(register Console * C)
 {
-    outw((C->basepage & 0xff00) | 0x0c, CCBase);
-    outw((C->basepage << 8) | 0x0d, CCBase);
+    outw((C->vseg_offset & 0xff00) | 0x0c, C->crtc_base);
+    outw((C->vseg_offset << 8) | 0x0d, C->crtc_base);
 }
 
 static void PositionCursor(register Console * C)
 {
-    unsigned int Pos = C->cx + Width * C->cy + C->basepage;
+    unsigned int Pos = C->cx + C->w * C->cy + C->vseg_offset;
 
-    outb(14, CCBase);
-    outb(Pos >> 8, CCBase + 1);
-    outb(15, CCBase);
-    outb(Pos, CCBase + 1);
+    outb(14, C->crtc_base);
+    outb(Pos >> 8, C->crtc_base + 1);
+    outb(15, C->crtc_base);
+    outb(Pos, C->crtc_base + 1);
 }
 
-static void DisplayCursor(int onoff)
+static void DisplayCursor(register Console * C, int onoff)
```

Comment:
Don't add new code with register declarations anywhere. They're completely ignored by GCC and are slowly being removed then a file is updated. I would recommend removing them all here, but the PR is large enough already.